packages/info/index.ts

```diff
@@ -1,22 +1,57 @@
+import chalk from "chalk";
 import * as envinfo from "envinfo";
 import * as process from "process";
+import { argv } from "./options";
 
-/**
- * Prints debugging information for webpack issue reporting
- */
-
-// TODO: define proper interface
-// eslint-disable-next-line
-export function information(): any {
-	return {
-		Binaries: ["Node", "Yarn", "npm"],
-		Browsers: ["Chrome", "Firefox", "Safari"],
-		System: ["OS", "CPU"],
-		npmGlobalPackages: ["webpack", "webpack-cli"],
-		npmPackages: "*webpack*"
-	};
+import { AVAILABLE_COMMANDS, AVAILABLE_FORMATS, IGNORE_FLAGS } from "./commands";
+
+const CONFIG = {};
+const DEFAULT_DETAILS = {
+	Binaries: ["Node", "Yarn", "npm"],
+	Browsers: ["Chrome", "Firefox", "Safari"],
+	System: ["OS", "CPU", "Memory"],
+	npmGlobalPackages: ["webpack", "webpack-cli"],
+	npmPackages: "*webpack*",
+};
+let DETAILS_OBJ = {};
+
+export function informationType(type: string): object {
+	switch (type) {
+		case "system":
+			return { System: ["OS", "CPU", "Memory"] };
+		case "binaries":
+			return { Binaries: ["Node", "Yarn", "npm"] };
+		case "browsers":
+			return { Browsers: ["Chrome", "Firefox", "Safari"] };
+		case "npmg":
+			return { npmGlobalPackages: ["webpack", "webpack-cli"] };
+		case "npm":
+			return { npmPackages: "*webpack*" };
+
+	}
 }
 
 export default async function info(): Promise<void> {
-	process.stdout.write(await envinfo.run(information()));
+
+	Object.keys(argv).forEach(async (flag): Promise<void> => {
+		if (IGNORE_FLAGS.includes(flag)) {
+			return;
+		} else if (AVAILABLE_COMMANDS.includes(flag)) {
+			const flagVal = informationType(flag);
+			DETAILS_OBJ = { ...DETAILS_OBJ, ...flagVal };
+		} else if (AVAILABLE_FORMATS.includes(flag)) {
+			CONFIG[flag] = true;
+		} else {
+			// Invalid option
+			process.stdout.write("\n" + chalk.bgRed(flag) + chalk.red(" is an invalid option"));
+			return;
```

Comment:
return here is unneeded as no more statement are left in callback. You instead exit the process at this point after printing.

Reply:
The `forEach` iterator iterates over the given array and after that the remaining instructions in the `info()` func gets executed.

Reply:
As the func is going to be imported, hence proper returning is required.
> Have a look at the new update

Reply:
Returning from callback is not going to return from info function, just from callback. Ending process here is better IMO.